### PR TITLE
Refactors dbcore and limits the maximum amount of concurrent async queries to a variable amount

### PR DESCRIPTION
**Pull request by Watermelon914**
## About The Pull Request

Refactors dbcore to work off a subsystem if executed async and limits the maximum amount of concurrent async queries to 25.

This has been tested locally on a mysql docker image and there were no crashes (as long as you didn't run it with debug extools) + data was getting recorded fine.

## Why It's Good For The Game

May or may not resolve terry crashes, however, each query creates a new thread which takes up 2mb, preventing the game from using that 2mb. This can lead to ooms if they stack up, e.g. due to poor connectivity. This solves that issue.

## Changelog
:cl:
refactor: Refactors dbcore.dm to possibly resolve the crashes that happen on Terry.
/:cl:

### CODE REVIEW

File: code/controllers/subsystem/dbcore.dm [hunk 1]
```diff
@@ -46,14 +127,27 @@ SUBSYSTEM_DEF(dbcore)
...
+	if(var_name == NAMEOF(src, queries_new))
+		return FALSE
+	if(var_name == NAMEOF(src, queries_standby))
+		return FALSE
+	if(var_name == NAMEOF(src, queries_active))
+		return FALSE
+
+	return ..()
 
 /datum/controller/subsystem/dbcore/vv_edit_var(var_name, var_value)
```
Comment (MrStonedOne) [inline code/controllers/subsystem/dbcore.dm:152]: I do not think vv_edit_var checks that the var was allowed to be gotten, and could still be edited by spoofing hrefs.

You likely need to add the same code down here.

**Comment by optimumtact** [comment]
Could you please add a debug verb that lists the number of queries held, the number sent to rust-g etc?

Some statistics would be nice to give us a sense of how things are going.

**Comment by Time-Green** [comment]
Terry hasn’t crashed for two rounds in a row since this got testmerged, could this be it?

**Comment by Watermelon914** [comment]
> Terry hasn’t crashed for two rounds in a row since this got testmerged, could this be it?

not yet.

**Comment by Time-Green** [comment]
This actually fixed the Terry crashes, god bless

**Comment by optimumtact** [comment]
It didn't, but it was helpful

**Pull request merged**